astropy/time/core.py

```diff
@@ -1863,12 +1863,23 @@ def sidereal_time(self, kind, longitude=None, model=None):
 
         Returns
         -------
-        sidereal time : `~astropy.coordinates.Longitude`
-            Sidereal time as a quantity with units of hourangle
+        `~astropy.coordinates.Longitude`
+            Sidereal time or Earth rotation angle, with units of hourangle.
 
         See Also
         --------
         astropy.time.Time.earth_rotation_angle : short-cut for ``kind='era'``.
+
+        Notes
+        -----
+        Sidereal time is referred to the equinox and is a complicated function of
+        both UT1 and Terrestrial Time. Earth rotation angle is a more modern
+        substitute. It is referred to the Celestial Intermediate Origin (CIO) and
+        is a linear function of UT1 alone.
```

Comment:
I would add the following here:

The difference between apparent sidereal time and Earth rotation angle is the equation of the origins which is the angle between the equinox and the CIO. Applying apparent sidereal time to the hour angle yields the true apparent Right Ascension with respect to the equinox, while applying the Earth rotation angle yields the intermediate Right Ascension with respect to the CIO.

Reply:
In this regard, Earth rotation angle isn't really a substitute for sidereal time. In some sense, it's just a more modern and convenient method of calculating sidereal time by first calculating the Earth rotation angle and then subtracting the equation of the origins as is done in `erfa.gst06()`.

Reply:
Nice, done.